### PR TITLE
feat(a380x/fadec): N1 fan protection measures (METOTS, KOZ)

### LINKED ISSUE #9283: Implement METOTS system for A380X
### Aircraft Version

Stable

### Description

When taking off, the FADEC limits the maximum thrust of the engines to 78% N1 until speed is above 35 knots to prevent flutter on the engine's front fan blades.

### References (optional)

https://www.youtube.com/watch?v=8d5q-J6ZEwo

### Additional info (optional)

I would like to help work on and implement such a system into the aircraft. Please contact me on discord if you're working on the engine model, I'd love to collaborate.

### Discord Username (optional)

tankman69

### PR DESCRIPTION
**Pull request by flogross89**
Fixes #9283 

## Summary of Changes

Implements METOTS and KOZ (keep-out zone) in the N1 control loop within the Autothrust.
Additionally, adds temporary provisions to draw the grey area (avail thrust) only until the METOTS area below CAS of 35kts. Once we have proper FADEC, this should be replaced by SimVars communicating the current min. and max. avail thrust. 

The reverse thrust limit changes are from #9417 

## Additional context

image

image

Discord username (if different from GitHub):

## Testing instructions

1) Perform TO/GA takeoff, below 35kts the thrust should be limited to 78% N1 max.
2) Perform FLX takeoff, below 35kts the thrust should be limited to 78% N1 max.
4) Ensure that the full range of thrust is available in normal flight

Below 35kts, it shouldn't be possible to command N1 > 78%
Below 60kts, it shouldn't be possible to command N1 between 64% and 72%

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page

### CODE REVIEW

**Comment by Tankman61** [comment]
I've just been testing myself and it looks good, just a few small things to ask. I don't know if they really matter/are intentional/fixable but here they are: 

- Whenever I spawn into the aircraft on the runway and push the throttle levers to TOGA I'm able to bring the engines to 100% N1 temporarily before they go back down to 76.5% N1
- The grey area is slightly greater than the maximum commanded N1 possible (76.5%) 
- I'm able to g et the engines to above 76.5% N1 but no greater than 78% N1 when I put the thrust levers between climb and idle while under 35 knots of airspeed

I can upload screenshots if needed but these are relatively minor

**Comment by flogross89** [comment]
@Tankman61 Thank you! Could you please provide a video for 1) and 3)? Maybe there's a condition or case I missed.

**Comment by Tankman61** [comment]
Here's a screenshot for 3, I have a video on another device which I will upload when I get home but it was really easy to reproduce, all you do is spawn in on a runway at an airport and immediately set the thrust levers to toga. As the engines spool up they will reach 100% N1 for a few seconds before reducing back to 76.5% N1.
Screenshot 2024-11-11 145520

**Comment by flogross89** [comment]
@Tankman61 Thanks! This only happens when spawning on the runway and setting TOGA immediately, right? We likely have some internal confirmation circuits for the ground condition and some startup timing stuff, that's why it probably takes 5-10s after startup of all components to behave correctly. This is expected, as this use case of spawning and insta-TOGA is highly unrealistic :)

**Comment by alexr4339** [comment]
Quality Assurance Tester

Discord Username  : alexdnzdmb
Object of testing : #9481
Aircraft          : A380X
Tier of Testing   : 1
Date              : 15/11/2024

Testing Process: 
Perform TO/GA takeoff, below 35kts the thrust should be limited to 78% N1 max. - working, N1 around 76,5%
Perform FLX takeoff, below 35kts the thrust should be limited to 78% N1 max. - working, N1 around 76,5%
Ensure that the full range of thrust is available in normal flight - it is
Below 35kts, it shouldn't be possible to command N1 > 78% - working, N1 around 76,5%
Below 60kts, it shouldn't be possible to command N1 between 64% and 72% - working, N1 around 62,5% so not right at the edge

Testing Results: 
Passed

Negatives:       
None

Conclusions: 
Great addition

Media:

File: fbw-a380x/src/systems/instruments/src/EWD/elements/ThrustGauge.tsx [hunk 1]
```diff
@@ -394,8 +420,8 @@ export class ThrustGauge extends DisplayComponent<ThrustGaugeProps> {
             <GaugeThrustComponent
               x={this.props.x}
               y={this.props.y}
-              valueIdle={0.04}
-              valueMax={2.6}
+              valueIdle={Subject.create(0.04)}
```
Comment (lukecologne) [inline fbw-a380x/src/systems/instruments/src/EWD/elements/ThrustGauge.tsx:423]: Not sure if this is good practice or not in the avionics framework, maybe someone else can comment on that

**Review by lukecologne — APPROVED** [review]

File: fbw-a380x/src/systems/instruments/src/EWD/elements/ThrustGauge.tsx (hunk 1 above)
Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/EWD/elements/ThrustGauge.tsx:423]: improved in latest version, after input from tracer :)

**Pull request merged**